extension/services/spotify/player.js

```diff
@@ -18,12 +18,21 @@ this.player = (function() {
       );
       this.setReactInputValue(input, query);
       if (thenPlay) {
-        const playerButton = await this.waitForSelector(SEARCH_PLAY, {
-          timeout: 2000,
-          // There seem to be 3 fixed buttons that appear early before the search results
-          minCount: 4,
-        });
-        playerButton.click();
+        try {
+          const playerButton = await this.waitForSelector(SEARCH_PLAY, {
+            timeout: 2000,
+            // There seem to be 3 fixed buttons that appear early before the search results
+            minCount: 4,
+          });
+          playerButton.click();
+        } catch (e) {
+          if (e.name === "TimeoutError") {
+            const e = new Error("No search results");
+            e.name = "EmptyResults";
+            e.displayMessage = `No results found for ${query}`;
```

Comment:
Oh... I think I pointed you the wrong way here. The exception is getting back, but `displayMessage` is getting lost. And right now we don't have a good way to keep it when the exception is passed from the content script (where `player.js` runs) to the background page (where `spotify.js` is running).

I'd recommend just throwing `new Error("No search results")` here and then catch it in `spotify.js`: https://github.com/mozilla/firefox-voice/blob/master/extension/services/spotify/spotify.js#L9 and add `.displayMessage` to it there. Something like:

```js
try {
  await this.callTab(...)
} catch (e) {
  if (e.message.includes("No search results")) {
    e.displayMessage = `...`
  }
  throw e;
}
```